src/main/java/games/strategy/engine/framework/map/download/MapDownloadProgressListener.java

```diff
@@ -0,0 +1,59 @@
+package games.strategy.engine.framework.map.download;
+
+import java.util.Optional;
+
+import javax.swing.JProgressBar;
+import javax.swing.SwingUtilities;
+
+/**
+ * A listener of map download progress events that updates the associated controls in the UI.
+ *
+ * <p>
+ * Instances of this class are thread safe.
+ * </p>
+ */
+final class MapDownloadProgressListener {
+  private static final int MIN_PROGRESS_VALUE = 0;
+
+  private static final int MAX_PROGRESS_VALUE = 100;
+
+  private final JProgressBar progressBar;
+
+  private final String uri;
+
+  private volatile Optional<Long> downloadLength = Optional.empty();
+
+  MapDownloadProgressListener(final String uri, final JProgressBar progressBar) {
+    this.progressBar = progressBar;
+    this.uri = uri;
+  }
+
+  void downloadStarted() {
+    SwingUtilities.invokeLater(() -> {
+      progressBar.setMinimum(MIN_PROGRESS_VALUE);
+      progressBar.setMaximum(MAX_PROGRESS_VALUE);
+      progressBar.setIndeterminate(true);
+      progressBar.setStringPainted(false);
+    });
+
+    downloadLength = DownloadUtils.getDownloadLength(uri);
```

Comment:
`downloadStarted()` is always called from a dedicated background thread, so it is acceptable to call a potentially long-running network operation (`DownloadUtils#getDownloadLength()`) here.